File: tests/Alkoholspiel.Core.Tests/FakeRepositories/FakeGameRepository.cs
using Alkoholspiel.Core.Entities;

namespace Alkoholspiel.Core.Tests.FakeRepositories;

public class FakeGameRepository : IGameRepository
{
    public List<Game> Data { get; set; }

    public FakeGameRepository()
    {
        Data = new List<Game>();
    }

    public Task<Game> Insert(Game game)
    {
        Data.Add(game);
        return Task.FromResult(game);
    }
}

File: core/src/test/kotlin/alkoholspiele/core/persistance/FakeGameRepository.kt
package alkoholspiele.core.persistance

import alkoholspiele.core.entities.Game
import java.util.*
import kotlin.collections.HashMap

class FakeGameRepository : GameRepository {

    val data: HashMap<UUID, Game> = HashMap();

    override suspend fun insert(game: Game) {
        data[game.id] = game;
    }

    override suspend fun getById(id: UUID): Game? {
        return data[id];
    }
}